netbox/core/forms/model_forms.py

```diff
@@ -88,14 +90,34 @@ class Meta:
         model = ManagedFile
         fields = ('data_source', 'data_file', 'auto_sync_enabled')
 
+    def _validate_file_extension(self, file):
+        if file:
+            extension = Path(file.name).suffix[1:].lower()
+
+            if extension != "py":
+                raise ValidationError(
+                    "File extension “%(extension)s” is not allowed .py extension is required",
+                    code="invalid_extension",
+                    params={
+                        "extension": extension,
+                    },
+                )
+
     def clean(self):
         super().clean()
 
-        if self.cleaned_data.get('upload_file') and self.cleaned_data.get('data_file'):
+        upload_file = self.cleaned_data['upload_file']
+        data_file = self.cleaned_data['data_file']
+
+        if upload_file and data_file:
+            extension = Path(upload_file.name).suffix[1:].lower()
             raise forms.ValidationError("Cannot upload a file and sync from an existing file")
-        if not self.cleaned_data.get('upload_file') and not self.cleaned_data.get('data_file'):
+        if not upload_file and not data_file:
             raise forms.ValidationError("Must upload a file or select a data file to sync")
 
+        self._validate_file_extension(upload_file)
+        self._validate_file_extension(data_file)
```

Comment:
Sorry, I think I gave you the wrong impression when we spoke about validation earlier. I meant that we could explore enforcing specific file suffixes, however we can't force that change as part of a bug fix; it would need to be proposed & implemented under a separate FR. The changes in `views.py` should be sufficient to resolve the bug.